### PR TITLE
Merge old and new FileDialogDesc

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 1]
```diff
@@ -353,38 +353,34 @@ static bool Browse(bool isSave, char* path, size_t pathSize)
             extension = ".park";
             fileType = FileExtension::PARK;
             title = isSave ? STR_FILE_DIALOG_TITLE_SAVE_GAME : STR_FILE_DIALOG_TITLE_LOAD_GAME;
-            desc.filters[0].name = language_get_string(STR_OPENRCT2_SAVED_GAME);
-            desc.filters[0].pattern = GetFilterPatternByType(_type, isSave);
+            desc.Filters.push_back({ language_get_string(STR_OPENRCT2_SAVED_GAME), GetFilterPatternByType(_type, isSave) });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/LoadSave.cpp:356]:
```suggestion
            desc.Filters.emplace_back(language_get_string(STR_OPENRCT2_SAVED_GAME), GetFilterPatternByType(_type, isSave));
```

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 2]
```diff
@@ -353,38 +353,34 @@ static bool Browse(bool isSave, char* path, size_t pathSize)
...
             break;
 
         case LOADSAVETYPE_LANDSCAPE:
             extension = ".park";
             fileType = FileExtension::PARK;
             title = isSave ? STR_FILE_DIALOG_TITLE_SAVE_LANDSCAPE : STR_FILE_DIALOG_TITLE_LOAD_LANDSCAPE;
-            desc.filters[0].name = language_get_string(STR_OPENRCT2_LANDSCAPE_FILE);
-            desc.filters[0].pattern = GetFilterPatternByType(_type, isSave);
+            desc.Filters.push_back({ language_get_string(STR_OPENRCT2_LANDSCAPE_FILE), GetFilterPatternByType(_type, isSave) });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/LoadSave.cpp:363]:
```suggestion
            desc.Filters.emplace_back(language_get_string(STR_OPENRCT2_LANDSCAPE_FILE), GetFilterPatternByType(_type, isSave));
```

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 3]
```diff
@@ -353,38 +353,34 @@ static bool Browse(bool isSave, char* path, size_t pathSize)
...
             break;
 
         case LOADSAVETYPE_SCENARIO:
             extension = ".park";
             fileType = FileExtension::PARK;
             title = STR_FILE_DIALOG_TITLE_SAVE_SCENARIO;
-            desc.filters[0].name = language_get_string(STR_OPENRCT2_SCENARIO_FILE);
-            desc.filters[0].pattern = GetFilterPatternByType(_type, isSave);
+            desc.Filters.push_back({ language_get_string(STR_OPENRCT2_SCENARIO_FILE), GetFilterPatternByType(_type, isSave) });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/LoadSave.cpp:370]:
```suggestion
            desc.Filters.emplace_back(language_get_string(STR_OPENRCT2_SCENARIO_FILE), GetFilterPatternByType(_type, isSave));
```

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 4]
```diff
@@ -353,38 +353,34 @@ static bool Browse(bool isSave, char* path, size_t pathSize)
...
             break;
 
         case LOADSAVETYPE_TRACK:
             extension = ".td6";
             fileType = FileExtension::TD6;
             title = isSave ? STR_FILE_DIALOG_TITLE_SAVE_TRACK : STR_FILE_DIALOG_TITLE_INSTALL_NEW_TRACK_DESIGN;
-            desc.filters[0].name = language_get_string(STR_OPENRCT2_TRACK_DESIGN_FILE);
-            desc.filters[0].pattern = GetFilterPatternByType(_type, isSave);
+            desc.Filters.push_back(
+                { language_get_string(STR_OPENRCT2_TRACK_DESIGN_FILE), GetFilterPatternByType(_type, isSave) });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/LoadSave.cpp:378]:
```suggestion
            desc.Filters.emplace_back(
                language_get_string(STR_OPENRCT2_TRACK_DESIGN_FILE), GetFilterPatternByType(_type, isSave));
```

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 5]
```diff
@@ -353,38 +353,34 @@ static bool Browse(bool isSave, char* path, size_t pathSize)
...
             break;
 
         case LOADSAVETYPE_HEIGHTMAP:
             title = STR_FILE_DIALOG_TITLE_LOAD_HEIGHTMAP;
-            desc.filters[0].name = language_get_string(STR_OPENRCT2_HEIGHTMAP_FILE);
-            desc.filters[0].pattern = GetFilterPatternByType(_type, isSave);
+            desc.Filters.push_back({ language_get_string(STR_OPENRCT2_HEIGHTMAP_FILE), GetFilterPatternByType(_type, isSave) });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/LoadSave.cpp:383]:
```suggestion
            desc.Filters.emplace_back(language_get_string(STR_OPENRCT2_HEIGHTMAP_FILE), GetFilterPatternByType(_type, isSave));
```

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 6]
```diff
@@ -409,21 +405,20 @@ static bool Browse(bool isSave, char* path, size_t pathSize)
         }
     }
 
-    desc.initial_directory = _directory;
-    desc.type = isSave ? FileDialogType::Save : FileDialogType::Open;
-    desc.default_filename = isSave ? path : nullptr;
+    desc.InitialDirectory = _directory;
+    desc.Type = isSave ? OpenRCT2::Ui::FileDialogType::Save : OpenRCT2::Ui::FileDialogType::Open;
+    desc.DefaultFilename = isSave ? path : "";
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/LoadSave.cpp:410]:
```suggestion
    desc.DefaultFilename = isSave ? path : std::string();
```

File: src/openrct2/config/Config.cpp [hunk 7]
```diff
@@ -718,20 +718,16 @@ namespace Config
...
-        desc.filters[0].name = language_get_string(STR_GOG_INSTALLER);
-        desc.filters[0].pattern = "*.exe";
-        desc.filters[1].name = language_get_string(STR_ALL_FILES);
-        desc.filters[1].pattern = "*";
-        desc.filters[2].name = nullptr;
+        FileDialogDesc desc{};
+        desc.Type = FileDialogType::Open;
+        desc.Title = language_get_string(STR_SELECT_GOG_INSTALLER);
+        desc.Filters.push_back({ language_get_string(STR_GOG_INSTALLER), "*.exe" });
+        desc.Filters.push_back({ language_get_string(STR_ALL_FILES), "*" });
```
Comment (tupaschoal) [inline src/openrct2/config/Config.cpp:725]:
```suggestion
        desc.Filters.emplace_back(language_get_string(STR_GOG_INSTALLER), "*.exe");
        desc.Filters.emplace_back(language_get_string(STR_ALL_FILES), "*");
```

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 8]
```diff
@@ -343,8 +343,8 @@ static void WindowLoadsaveResize(rct_window* w)
 
 static bool Browse(bool isSave, char* path, size_t pathSize)
 {
-    file_dialog_desc desc = {};
-    const utf8* extension = "";
+    OpenRCT2::Ui::FileDialogDesc desc = {};
+    u8string extension = "";
```
[the remark this answers is not in the document]
Comment (Gymnasiast) [reply src/openrct2-ui/windows/LoadSave.cpp:347]: Even though your version would work, it’s still massively distracting when you’re looking through the code to debug it. That’s why I prefer explicit initialisation.

**Pull request merged**